lib/store/src/StoryStore.ts

```diff
@@ -97,20 +99,28 @@ export class StoryStore<TFramework extends AnyFramework> {
     this.prepareStoryWithCache = memoize(STORY_CACHE_SIZE)(prepareStory) as typeof prepareStory;
   }
 
-  async initialize({ cacheAllCSFFiles = false }: { cacheAllCSFFiles?: boolean } = {}) {
-    await this.storiesList.initialize();
+  initialize(args: { sync: false; cacheAllCSFFiles?: boolean }): Promise<void>;
 
-    if (cacheAllCSFFiles) {
-      await this.cacheAllCSFFiles();
-    }
-  }
+  initialize(args: { sync: true; cacheAllCSFFiles?: boolean }): void;
 
-  initializeSync({ cacheAllCSFFiles = false }: { cacheAllCSFFiles?: boolean } = {}) {
-    this.storiesList.initializeSync();
-
-    if (cacheAllCSFFiles) {
-      this.cacheAllCSFFilesSync();
+  initialize({
+    sync,
```

Comment:
```suggestion
    sync = false,
```
Assuming that's allowed when overloading.